backend/api/parse_pdf.py

```
@@ -840,7 +1438,23 @@ async def parse_pdf_with_progress_route(
-    asyncio.create_task(parse_pdf_with_progress(file, context_files, save_to_db, session_id, db))
+    
+    async def run_parsing_with_error_handling():
+        """Wrapper to catch exceptions from background task"""
+        try:
+            await parse_pdf_with_progress(file, context_files, save_to_db, session_id, db, current_user)
+        except HTTPException as e:
+            # HTTPExceptions (like 504 timeout) should update progress and not crash
+            debug_log(f"[PROGRESS] HTTPException in background task: {e.status_code} - {e.detail}")
+            if session_id:
+                progress_manager.error_processing(session_id, f"{e.detail}")
+        except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)